### PR TITLE
Make PAM macros more flexible to variables

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

It may exist cases where variables are passed to these macros

- e.g.: `accounts_password_pam_pwhistory_remember_password_auth`

In some cases it these variables were not resolved in Bash macros and the Playbooks were breaking.
These macros can now properly use the variables.

#### Rationale:

More flexibility for PAM settings.

#### Review Hints:

These macros are used by many rules.
Nothing is expected to break with these changes, but lets check the CI tests.

Just for information: https://fedoramagazine.org/authselect-in-fedora-linux-40-migrating-to-the-new-local-profile/

### CODE REVIEW

**Comment by marcusburghardt** [comment]
I moved it to draft while I investigate the issues reported in CI tests.

**Review by jan-cerny — COMMENTED** [review]
I have run some Automatus test scenarios with both Bash and Ansible and they all look fine.

I love the great commit messages.

File: shared/macros/10-ansible.jinja [hunk 1]
```diff
@@ -1421,10 +1421,14 @@ Part of the grub2_bootloader_argument_absent template.
 
 #}}
 {{%- macro ansible_ensure_pam_module_line(pam_file, group, control, module, after_match='') -%}}
+- name: '{{{ rule_title }}} - Define a fact for control already filtered in case filters are used'
+  ansible.builtin.set_fact:
+    pam_module_control: '{{{ control }}}'
+
 - name: '{{{ rule_title }}} - Check if expected PAM module line is present in {{{ pam_file }}}'
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply shared/macros/10-ansible.jinja:1428]: Yes, the Ansible variable will be translated during the Playbook execution.

[the remark this answers is not in the document]
Comment (jan-cerny) [reply shared/macros/10-ansible.jinja:1428]: thanks

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**